gef.py

```diff
@@ -4933,25 +4933,25 @@ def context_code(self):
 
             lines = gef_disassemble(pc, nb_insn)
             for addr, content in lines:
-                insn = "%#x %s" % (addr,content)
-                line = u""
+                insn = "%#x %s" % (addr, content)
+                line = []
                 if addr < pc:
-                    line+= Color.grayify("%#x\t %s" % (addr, content,) )
+                    line+= Color.grayify("%#x\t%s" % (addr, content,) )
                 elif addr == pc:
-                    line+= Color.colorify("%#x\t %s \t\t %s $pc" % (addr, content, left_arrow), attrs="bold red")
+                    line+= Color.colorify("%#x\t%s\t\t%s $pc" % (addr, content, left_arrow), attrs="bold red")
 
                     if is_conditional_branch(insn):
                         is_taken, reason = is_branch_taken(insn)
-                        reason = "[Reason: %s]" % reason if len(reason) else ""
+                        reason = "[%s]" % reason if reason else ""
                         if is_taken:
-                            line+= Color.colorify("\tBranch TAKEN %s" % reason, attrs="bold green")
+                            line+= Color.colorify("\tTAKEN %s" % reason, attrs="bold green")
                         else:
-                            line+= Color.colorify("\tBranch NOT taken %s" % reason, attrs="bold red")
+                            line+= Color.colorify("\tNOT taken %s" % reason, attrs="bold red")
 
                 else:
-                    line+= "%#x\t %s" % (addr, content)
+                    line+= "%#x\t%s" % (addr, content)
 
-                print(line)
+                print(u"".join(line))
```

Comment:
This is very interesting!

You're right, my test was incomplete. Using yours, I noticed that the array version is **always** faster on Python2 (by ~16%), whereas the string and table version are approx. the same speed on Python3. 

So you're right, on average it is a bit faster to use the table approach. I will integrate this.

Thank you.